src/packages/database/model/utils/persistence.js

```diff
@@ -23,6 +23,24 @@ export function create(record: Model, trx: Object): Array<Object> {
     updatedAt: timestamp
   });
 
+  if (record.allowPrimaryKeyInsert) {
+    const columns = getColumns(record);
+    // check that we have a valid id here?
```

Comment:
I don't think that we have to check if the id is valid. If this fails due to an error that is thrown by the database driver that is ok. No need for us to add an additional error message and spend more time in node than what is necessary.